frontend/src/Frontend/UI/DeploymentSettings.hs

```
@@ -572,21 +579,87 @@ uiChainSelection info cls = mdo
+      ]
+    results :: Dynamic t (IntMap (Dynamic t Bool, Dynamic t (Maybe (Map AccountName [SigCapability]))))
+      <- foldDyn applyAlways im0 im'
+    let nextKeyToUse = maybe 0 (succ . fst) . IM.lookupMax <$> current results
+        canDelete = (> 1) . IM.size <$> current results
+        anyEmpty = fmap or $ traverse fst =<< results
+        deletions = switch . current $ IM.foldMapWithKey (\i -> (IM.singleton i Nothing <$) . ffilter id . updated . fst) <$> results
+  pure $ fmap (fmap (Map.unionsWith (<>) . IM.elems) . sequence) $ traverse snd =<< results
 
 -- | Widget for selection of sender and signing keys.
-uiSigningKeysSender
-  :: forall t m model. (MonadWidget t m, HasWallet model t)
-  => model
+uiSenderCapabilities
+  :: forall t m. MonadWidget t m
+  => Maybe [DappCap]
   -> m (Dynamic t (Maybe AccountName))
-  -> m (Dynamic t (Maybe AccountName), Dynamic t (Set KeyName))
-uiSigningKeysSender model mkSender = do
+  -> m (Dynamic t (Maybe AccountName), Dynamic t (Maybe (Map AccountName [SigCapability])))
+uiSenderCapabilities mCaps mkSender = do
   divClass "title" $ text "Gas Payer"
-
   sender <- divClass "group" mkSender
 
-  divClass "title" $ text "Transaction Signer"
-  keys <- divClass "group" $ uiSigningKeys model
-  return (sender, keys)
+  -- Capabilities
+  -- TODO consider hard coding a single "FUND_TX" / gas field instead of the
+  -- above gas payer input
+  divClass "title" $ text "Roles"
+  capabilities <- divClass "group" $ elAttr "table" ("class" =: "table" <> "style" =: "width: 100%") $ case mCaps of
+    Nothing -> el "tbody" $ capabilityInputRows mkSender
+    Just caps -> do
+      el "thead" $ el "tr" $ do
+        elClass "th" "table__heading" $ text "Role"
+        elClass "th" "table__heading" $ text "Capability"
+        elClass "th" "table__heading" $ text "Account"
+      let fixup = fmap (fmap (Map.unionsWith (<>)) . sequence) . sequence
```

What does the `Maybe` in `Maybe AccountName` mean?

IIUC, the inner `sequence` is 
```
   [Maybe (Map AccountName [SigCapability])] 
-> Maybe [(Map AccountName [SigCapability])]
```
so any `Nothing` in the list will cause the whole thing to be `Nothing`

Why do we want that rather than dropping the failures, e.g. 
```
fmapMaybe id 
:: [Maybe (Map AccountName [SigCapability])] 
-> [(Map AccountName [SigCapability])]
```

---

I didn't want to allow submission of a TX by silently dropping partial capabilities